### ISSUE TITLE
When shutting down Kamanja, SmartFileConsumer throws exception for each directory monitored over SFTP

### ISSUE THREAD

**Issue by ligadata-william** [LigaData/Kamanja#1517]
In my case, I've configured Kamanja to use the SmartFileConsumer to monitor two directories on a remote machine using SFTP. When Kamanja shuts down, a java.io.InterruptedIOException causes an SftpException. This is possibly because Kamanja doesn't properly close off the SFTP pipe before shutting down. This should be cleaned up. See exceptions below:
```
ERROR [pool-476-thread-1] - Sftp File Handler - Error while listing files of dir /home/william/data/helloworld/input2
com.jcraft.jsch.SftpException:
	at com.jcraft.jsch.ChannelSftp.ls(ChannelSftp.java:1720) ~[ExtDependencyLibs2_2.11-1.6.2.jar:1.6.2]
	at com.jcraft.jsch.ChannelSftp.ls(ChannelSftp.java:1526) ~[ExtDependencyLibs2_2.11-1.6.2.jar:1.6.2]
	at com.ligadata.InputAdapters.sftp.SftpFileHandler.listDirectFiles(SftpFileHandler.scala:554) [KamanjaInternalDeps_2.11-1.6.2.jar:1.6.2]
	at com.ligadata.InputAdapters.sftp.SftpFileHandler.listFiles(SftpFileHandler.scala:597) [KamanjaInternalDeps_2.11-1.6.2.jar:1.6.2]
	at com.ligadata.InputAdapters.MonitorController.com$ligadata$InputAdapters$MonitorController$$monitorBufferingFiles(MonitorController.scala:526) [KamanjaInternalDeps_2.11-1.6.2.jar:1.6.2]
	at com.ligadata.InputAdapters.MonitorController$$anonfun$monitor$2$$anon$3.run(MonitorController.scala:239) [KamanjaInternalDeps_2.11-1.6.2.jar:1.6.2]
	at scala.actors.threadpool.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1060) [ExtDependencyLibs_2.11-1.6.2.jar:1.6.2]
	at scala.actors.threadpool.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:574) [ExtDependencyLibs_2.11-1.6.2.jar:1.6.2]
	at java.lang.Thread.run(Thread.java:745) [?:1.8.0_101]
Caused by: java.io.InterruptedIOException
	at java.io.PipedInputStream.read(PipedInputStream.java:328) ~[?:1.8.0_101]
	at java.io.PipedInputStream.read(PipedInputStream.java:377) ~[?:1.8.0_101]
	at com.jcraft.jsch.ChannelSftp.fill(ChannelSftp.java:2882) ~[ExtDependencyLibs2_2.11-1.6.2.jar:1.6.2]
	at com.jcraft.jsch.ChannelSftp.header(ChannelSftp.java:2908) ~[ExtDependencyLibs2_2.11-1.6.2.jar:1.6.2]
	at com.jcraft.jsch.ChannelSftp.checkStatus(ChannelSftp.java:2446) ~[ExtDependencyLibs2_2.11-1.6.2.jar:1.6.2]
	at com.jcraft.jsch.ChannelSftp._sendCLOSE(ChannelSftp.java:2465) ~[ExtDependencyLibs2_2.11-1.6.2.jar:1.6.2]
	at com.jcraft.jsch.ChannelSftp.ls(ChannelSftp.java:1697) ~[ExtDependencyLibs2_2.11-1.6.2.jar:1.6.2]
	... 8 more
ERROR [pool-476-thread-2] - Sftp File Handler - Error while listing files of dir /home/william/data/helloworld/input1
com.jcraft.jsch.SftpException:
	at com.jcraft.jsch.ChannelSftp.ls(ChannelSftp.java:1720) ~[ExtDependencyLibs2_2.11-1.6.2.jar:1.6.2]
	at com.jcraft.jsch.ChannelSftp.ls(ChannelSftp.java:1526) ~[ExtDependencyLibs2_2.11-1.6.2.jar:1.6.2]
	at com.ligadata.InputAdapters.sftp.SftpFileHandler.listDirectFiles(SftpFileHandler.scala:554) [KamanjaInternalDeps_2.11-1.6.2.jar:1.6.2]
	at com.ligadata.InputAdapters.sftp.SftpFileHandler.listFiles(SftpFileHandler.scala:597) [KamanjaInternalDeps_2.11-1.6.2.jar:1.6.2]
	at com.ligadata.InputAdapters.MonitorController.com$ligadata$InputAdapters$MonitorController$$monitorBufferingFiles(MonitorController.scala:526) [KamanjaInternalDeps_2.11-1.6.2.jar:1.6.2]
	at com.ligadata.InputAdapters.MonitorController$$anonfun$monitor$2$$anon$3.run(MonitorController.scala:239) [KamanjaInternalDeps_2.11-1.6.2.jar:1.6.2]
	at scala.actors.threadpool.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1060) [ExtDependencyLibs_2.11-1.6.2.jar:1.6.2]
	at scala.actors.threadpool.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:574) [ExtDependencyLibs_2.11-1.6.2.jar:1.6.2]
	at java.lang.Thread.run(Thread.java:745) [?:1.8.0_101]
Caused by: java.io.InterruptedIOException
	at java.io.PipedInputStream.read(PipedInputStream.java:328) ~[?:1.8.0_101]
	at java.io.PipedInputStream.read(PipedInputStream.java:377) ~[?:1.8.0_101]
	at com.jcraft.jsch.ChannelSftp.fill(ChannelSftp.java:2882) ~[ExtDependencyLibs2_2.11-1.6.2.jar:1.6.2]
	at com.jcraft.jsch.ChannelSftp.header(ChannelSftp.java:2908) ~[ExtDependencyLibs2_2.11-1.6.2.jar:1.6.2]
	at com.jcraft.jsch.ChannelSftp.checkStatus(ChannelSftp.java:2446) ~[ExtDependencyLibs2_2.11-1.6.2.jar:1.6.2]
	at com.jcraft.jsch.ChannelSftp._sendCLOSE(ChannelSftp.java:2465) ~[ExtDependencyLibs2_2.11-1.6.2.jar:1.6.2]
	at com.jcraft.jsch.ChannelSftp.ls(ChannelSftp.java:1697) ~[ExtDependencyLibs2_2.11-1.6.2.jar:1.6.2]
	... 8 more
```

**Comment by ligadata-pokuri** [comment]
While shutting down the adapters, we interrupt the current monitoring. So, we get Interrupt IO exception. This does not cause any issues. May be we can ignore throwing exception.

**Comment by ligadata-pokuri** [comment]
This is fixed in branch branch 1.6.2.Test.gitissue.1517

**Comment by ligadata-william** [comment]
Verified as fixed. Closing.
